lib/choria/colt/data_structurer.rb

```diff
@@ -0,0 +1,28 @@
+module Choria
+  class Colt
+    module DataStructurer
+      def self.structure(results)
+        results.map do |res|
+          # data.stdout seems to always be JSON, so parse it once.
+          res[:result] = JSON.parse res[:data][:stdout]
```

Comment:
According to https://github.com/puppetlabs/puppetlabs-ruby_task_helper/pull/20 stdout is either the serialization of a `Hash#to_json` or a `Object#to_s`.  So in some cases the JSON parsing will not be possible, i.e. when a task return something that when `#to_s` is called produce an invalid JSON String (e.g. `"["`, `[{'a' => 2}]`, etc).

Reply:
Any task I tried was JSON parsable, even `exec`, so do you have an example?

Reply:
Ah, we are at the choria level so yes, a raw string has been wrapped in a hash and serialized to JSON! You can use `res.bolt_task_result` since [2.26.0](https://github.com/choria-io/mcorpc-ruby-support/releases/tag/2.26.0) which was added to do this:

https://github.com/choria-io/mcorpc-ruby-support/pull/178/commits/246703a3683d432869cccef65fdcf5c570267caf

Reply:
I will use data from raw `res` in future to aggregate some metrics, so thanks for the helper, but I think `res` have some interesting keys.